src/twisted/internet/task.py

```diff
@@ -369,61 +381,56 @@ class CooperativeTask:
     @ivar _cooperator: the L{Cooperator} that this L{CooperativeTask}
         participates in, which is used to re-insert it upon resume.
 
-    @ivar _deferreds: the list of L{defer.Deferred}s to fire when this task
+    @ivar _deferreds: the list of L{Deferred}s to fire when this task
         completes, fails, or finishes.
 
-    @type _deferreds: C{list}
-
-    @type _cooperator: L{Cooperator}
-
     @ivar _pauseCount: the number of times that this L{CooperativeTask} has
         been paused; if 0, it is running.
 
-    @type _pauseCount: C{int}
-
     @ivar _completionState: The completion-state of this L{CooperativeTask}.
         L{None} if the task is not yet completed, an instance of L{TaskStopped}
         if C{stop} was called to stop this task early, of L{TaskFailed} if the
         application code in the iterator raised an exception which caused it to
         terminate, and of L{TaskDone} if it terminated normally via raising
         C{StopIteration}.
-
-    @type _completionState: L{TaskFinished}
     """
 
-    def __init__(self, iterator, cooperator):
+    def __init__(
+        self, iterator: Iterator[_TaskResultT], cooperator: "Cooperator"
+    ) -> None:
         """
         A private constructor: to create a new L{CooperativeTask}, see
         L{Cooperator.cooperate}.
         """
         self._iterator = iterator
         self._cooperator = cooperator
-        self._deferreds = []
+        self._deferreds: List[Deferred[Iterator[_TaskResultT]]] = []
         self._pauseCount = 0
-        self._completionState = None
-        self._completionResult = None
+        self._completionState: Optional[SchedulerError] = None
+        self._completionResult: Optional[Union[Iterator[_TaskResultT], Failure]] = None
         cooperator._addTask(self)
 
-    def whenDone(self):
+    def whenDone(self) -> Deferred[Iterator[_TaskResultT]]:
         """
-        Get a L{defer.Deferred} notification of when this task is complete.
+        Get a L{Deferred} notification of when this task is complete.
 
-        @return: a L{defer.Deferred} that fires with the C{iterator} that this
+        @return: a L{Deferred} that fires with the C{iterator} that this
             L{CooperativeTask} was created with when the iterator has been
             exhausted (i.e. its C{next} method has raised C{StopIteration}), or
             fails with the exception raised by C{next} if it raises some other
             exception.
 
-        @rtype: L{defer.Deferred}
+        @rtype: L{Deferred}
```

Comment:
do we need this?

Reply:
Yes, since we aren't importing `defer`. Also, I think it's more readable that way.